### ISSUE TITLE
Commercial Qiqqa - Can't Sync

### ISSUE THREAD

**Issue by murilo992010** [jimmejardine/qiqqa-open-source#336]
I came to know Qiqqa recently. Didn't know anything about it, so I just downloaded the commercial version that I saw on Google.

After working for a few days with my documents, alternating between my own home and my GF's home and syncing when moving places, today I noticed that a "monthly free syncing limit" was reached (something like that). I would gladly pay the price to sync the damn thing and have access to my precious documents, so I can resume my research, but apparently, there is no such service available anymore. It was only after this happened that I discovered that Commercial Qiqqa has been discontinued and is now open source.

Anyway, I'm kind of lost about this and I don't know how to proceed. Any help/insight would be very much appreciated. Thank you!

(I'm at my house now, and the up-to-date files are in my GF's PC, at her house)

**Comment by GerHobbelt** [comment]
Since your work is important I suggest you create a backup of both machines' libraries before setting up any mechanism to sync both nodes later on. (Open Source Qiqqa has a couple of bugs in that department, so this is fair warning for more reasons than just making you do not loose anything in the transition. 😉 )

Personally, for this I would down the latest open source release (even while it's marked 'test/unpublished') and install it over the existing commercial Qiqqa install (which is most probably the last commercial release: v79).

FYI: all Qiqqa releases from commercial v79 up to and including the Open Source Qiqqa v83.XXX releases are all binary compatible, so no worries there.

When you have installed Qiqqa v83 like that (it should yak a bit about installing over an existing install; just click to continue then: this overwrites the Qiqqa **installed software** but your precious libraries are kept intact as they stored elsewhere on your system) it's time to create a Qiqqa backup file on both machines.

> The reason I advise backing up **both machines** is the fact that I have some suspicions about the way Qiqqa synchronizes multiple nodes under circumstances that in other systems produce "conflicts" (e.g. delete a document in one node; augment its metadata on another node: should the automated sync give the edit or the delete precedence, and questions in that vein) -- a scenario that's highly improbable in your case but with important stuff I'm loath to take risks. Hence dual backup before we go on.

Now on each machine where latest v83 has been installed, you should see the software automatically discovering your libraries again as before on the local box.

Then you can go to the sync manager and edit the 'sync location' to point at a directory where you can share data between the machines. 

> ## Sync Points 
>
> There are several ways to share data between machines when using Qiqqa:
>
> - using a directory on a NAS drive that's read+write accessible from all machines that need to share data.
> - using the "cloud" available publicly: while this is little more hairy (see also other issues in this issue tracker) you **could try** using a Google Drive directory ("folder") or DropBox-shared "folder".
> - using a USB thumbdrive or other device which you carry between machines: as long as you point each Qiqqa to the directory where the 'synchronized data' is stored, you can transport it back & forth between machines.
>
> **Notes**: Google Drive shares and DropBox in particular have caused trouble with Sync in the past. This has to do with Qiqqa using a SQLite database and the SQLite software sometimes doesn't like to write to "network attached storage" like that, resulting in error 14 and other trouble. YMMV. (A work-around for this is known, but requires some technical know-how and is out of scope for this message)
>

A screenshot or two to help you along the way towards that Sync Manager dialog in Qiqqa as it's the easiest spot to find the first time:

![image](https://user-images.githubusercontent.com/402462/123514379-09ed1100-d693-11eb-93c8-0b61b10dde76.png)

![image](https://user-images.githubusercontent.com/402462/123514426-4a4c8f00-d693-11eb-832c-1a229c82ecbd.png)

detail of last screenshot:

![image](https://user-images.githubusercontent.com/402462/123514438-5a646e80-d693-11eb-8b79-f4f61073a58a.png)

Click on Browse button to change directory: point it to where you have created the (still empty) directory, which will serve as Sync Point for this Qiqqa, e.g. a folder on a USB drive or Network drive (in my case, I'm using drive G: for that; it's an external SSD I use):

![image](https://user-images.githubusercontent.com/402462/123514519-b5966100-d693-11eb-9e5d-10fc3d496822.png)

![image](https://user-images.githubusercontent.com/402462/123514539-cfd03f00-d693-11eb-92d6-2621004175aa.png)

(just after I created the "New Folder" and clicked to select it. Now OK by click on "Select Folder":

![image](https://user-images.githubusercontent.com/402462/123514575-f8f0cf80-d693-11eb-80ec-9f70677f424d.png)

Notice that the listed sync point directory has changed we just picked.

Now you can run Sync to have Qiqqa synchronize your existing library (the one we ticked as in my example screenshots; don't mind the odd name of "TargetLib" I used there: I'm testing other Qiqqa so this library's name has no relevance)

![image](https://user-images.githubusercontent.com/402462/123514628-38b7b700-d694-11eb-90f0-060f0d8669a5.png)

Start Sync.

Qiqqa will start to work and in its status bar you should stuff like this happening:

![image](https://user-images.githubusercontent.com/402462/123514649-538a2b80-d694-11eb-9690-024306af9558.png)


Now you can carry this over to the other Qiqqa machine and point that one to this same Sync directory using the same procedure: when you hit Start Sync there Qiqqa should discover there's already a "remote copy" present now and only up/download the differences.

Validate/Check this by checking if your latest changes have made it to the other machine (in your current scenario, that would be your home box).

----

It doesn't (*shouldn't*) matter if you start with the machine where your did your latest work or the "older" one.

Again, make sure to check expected results. Qiqqa only synchronizes when instructed, so your work over at your GF only makes it to your home box when synced the GF machine to the share, *then* synced the home box, so it can the new data on there and pull it down.


----

Just in case: If reality doesn't match this description (errors? mistakes?), that's what the backups are for. 😉   Better safe than sorry.



--------

Take your time, and if any surprises pop up, please report them here. (Either in this issue on in new issues; you can easily reference an issue by typing `#336` (this issue's number) in your issue message and github should magically link to this one.

**Comment by GerHobbelt** [comment]
Related material:

- where to locate your Qiqqa libraries (local work data used by Qiqqa; **not** the sync point of course!) : https://github.com/jimmejardine/qiqqa-open-source/blob/master/docs-src/FAQ/How%20to%20locate%20your%20Qiqqa%20Base%20Directory.md
- installing qiqqa / updating qiqqa : https://github.com/jimmejardine/qiqqa-open-source/blob/master/docs-src/FAQ/Installing%20Qiqqa%20-%20Updating%20Qiqqa.md
- #257
